### PR TITLE
internal: use different schemes for the custom views

### PR DESCRIPTION
**Pull request by lnicola**
Related to #12699, but doesn't fix it because we still register the providers multiple times.

### CODE REVIEW

File: editors/code/src/commands.ts [hunk 1]
```diff
@@ -336,7 +336,7 @@ export function toggleInlayHints(_ctx: Ctx): Cmd {
 // The contents of the file come from the `TextDocumentContentProvider`
 export function syntaxTree(ctx: Ctx): Cmd {
     const tdcp = new (class implements vscode.TextDocumentContentProvider {
-        readonly uri = vscode.Uri.parse("rust-analyzer://syntaxtree/tree.rast");
+        readonly uri = vscode.Uri.parse("rust-analyzer-syntax-tree://syntaxtree/tree.rast");
```
Comment (bjorn3) [inline editors/code/src/commands.ts:339]: Maybe change the uri's into something like this?
```suggestion
        readonly uri = vscode.Uri.parse("rust-analyzer-syntax-tree://tree.rast");
```

Comment (lnicola) [reply editors/code/src/commands.ts:339]: I think we should go the other way around and have a single scheme, but sure.

Comment (lnicola) [reply editors/code/src/commands.ts:339]: For some reason this breaks the tab title (it shows `.` instead of `tree.rast` or whatever).

**Pull request merged**